IPython/core/interactiveshell.py

```diff
@@ -2641,13 +2758,33 @@ def error_before_exec(value):
         # compiler
         compiler = self.compile if shell_futures else CachingCompiler()
 
+        _run_async = False
+
         with self.builtin_trap:
             cell_name = self.compile.cache(cell, self.execution_count)
 
             with self.display_trap:
                 # Compile to bytecode
                 try:
-                    code_ast = compiler.ast_parse(cell, filename=cell_name)
+                    if _should_be_async(cell) and self.autoawait:
```

Comment:
I'd reverse these since autoawait is a bool and `_should_be_async` doesn't need to parse if the bool is False.